packages/commercetools/theme/components/LoginModal.vue

```diff
@@ -25,104 +25,142 @@
+                  class="form__input"
+                />
+              </ValidationProvider>
+              <ValidationProvider rules="required" v-slot="{ errors }">
+                <SfInput
+                  v-model="form.lastName"
+                  :valid="!errors[0]"
+                  :errorMessage="errors[0]"
+                  name="last-name"
+                  label="Last Name"
+                  class="form__input"
+                />
+              </ValidationProvider>
+              <ValidationProvider rules="required" v-slot="{ errors }">
+                <SfInput
+                  v-model="form.password"
+                  :valid="!errors[0]"
+                  :errorMessage="errors[0]"
+                  name="password"
+                  label="Password"
+                  type="password"
+                  class="form__input"
+                />
+              </ValidationProvider>
+              <SfCheckbox
+                v-model="createAccount"
+                name="create-account"
+                label="I want to create an account"
+                class="form__checkbox"
+              />
+              <SfButton
+                type="submit"
+                class="sf-button--full-width form__button"
+                :disabled="loading"
+              >
+                <SfLoader :class="{ loader: loading }" :loading="loading">
+                  <div>Create an account</div>
+                </SfLoader>
+              </SfButton>
+            </form>
+          </ValidationObserver>
           <div class="action">
             or
-            <SfButton class="sf-button--text" @click="isLogin = true"
-              >login in to your account</SfButton
-            >
+            <SfButton class="sf-button--text" @click="isLogin = true">login in to your account</SfButton>
           </div>
         </div>
       </transition>
     </SfModal>
   </div>
 </template>
 <script>
-import { SfModal, SfInput, SfButton, SfCheckbox } from '@storefront-ui/vue'
+import { computed, ref } from '@vue/composition-api'
+import { SfModal, SfInput, SfButton, SfCheckbox, SfLoader } from '@storefront-ui/vue'
+import { ValidationProvider, ValidationObserver, extend } from 'vee-validate'
+import { required, email } from 'vee-validate/dist/rules'
+import { useUser } from '@vue-storefront/commercetools-composables'
 import uiState from '~/assets/ui-state'
+
 const { isLoginModalOpen, toggleLoginModal } = uiState
 
+extend('email', {
+  ...email,
+  message: 'Invalid email'
+})
+
+extend('required', {
+  ...required,
+  message: 'This field is required'
+})
+
 export default {
   name: 'LoginModal',
-  components: { SfModal, SfInput, SfButton, SfCheckbox },
+  components: {
+    SfModal,
+    SfInput,
+    SfButton,
+    SfCheckbox,
+    SfLoader,
+    ValidationProvider,
+    ValidationObserver
+  },
   setup() {
+    const form = ref({})
+    const isLogin = ref(false)
+    const createAccount = ref(false)
+    const rememberMe = ref(false)
```

Comment:
imho it should be a param of login form and sent via `login`, wdyt?

Reply:
it is, it displays only on the login page, I used separated field because there are only fields that I needed to the registering - next task is for login I'll handle that